### PR TITLE
feat(fb): add form service tests, ci and deps

### LINKED ISSUE #88: feat(fb): establish form service test workflow parity with mail/user
## Description

Establish a **first-class test workflow for `services/form`** that matches the existing standards used in `services/mail` and `services/user` (local dev workflow, dependency pattern, test structure, Make targets, and CI workflow).

Right now `form` has no test suite and no test/dev dependency setup in `pyproject.toml`, while `mail` and `user` already have complete pytest + coverage + CI pipelines.

## Problem

The form service currently lacks the same testing workflow/setup/deps as other backend services, which creates uneven reliability across the monorepo:

- `services/form/pyproject.toml` has no test/dev extras (no `pytest`, `pytest-asyncio`, `pytest-cov`, etc.)
- `services/form` has no `tests/` directory
- root `Makefile` has no `test-form-service` target
- there is no dedicated CI workflow for form service tests (unlike `test-mail-service.yaml` and `test-user-service.yaml`)

As a result, regressions in form creation, updates, submissions, and validation logic can ship without service-level automated checks.

## Scope

Create parity with `mail` and `user` by implementing the same style of setup for `form`:

- dependency workflow (`uv sync --all-extras`)
- pytest config in `pyproject.toml`
- test folder organization and fixtures
- make target for running tests with coverage
- dedicated GitHub Actions test workflow

### 1) Add test/dev dependencies and pytest config to form service

Update `services/form/pyproject.toml` to include `project.optional-dependencies.dev` consistent with other backend services, including:

- `pytest`
- `pytest-asyncio`
- `pytest-cov`
- `mongomock-motor`
- `httpx` (if needed for API client tests)
- `ruff` (if we keep service-level parity with others)

Also add:

- `[tool.pytest.ini_options]`
- `asyncio_mode = "auto"`
- `testpaths = ["tests"]`
- `asyncio_default_fixture_loop_scope = "function"`

### 2) Add form test suite skeleton with same structure conventions

Create `services/form/tests/` with at least:

- `conftest.py` (settings patching, Mongo mocks, async/sync API clients)
- `test_api_health.py`
- `test_api_forms.py`
- `test_api_submissions.py`
- `test_models.py`
- `test_form_service.py`
- `test_submission_service.py`

Fixture approach should mirror `services/mail/tests/conftest.py` and `services/user/tests/conftest.py` patterns:

- patched service settings
- patched `MongoDB` connect/close for app lifespan tests
- reusable async cursor helper for mocked list queries

### 3) Define minimum behavioral coverage targets for first pass

Cover critical form flows and validation rules first:

- `POST /forms` happy path + validation failures
- `GET /forms/{id}` found / not found / invalid id
- `GET /forms` pagination and filtering behaviors
- `PUT /forms/{id}` partial update + invalid/not-found
- `DELETE /forms/{id}` success/not-found/invalid id
- `POST /submissions` success + form state validation failures
- submission retrieval endpoints (by id and by form)
- model validations for form schema and submission payloads

### 4) Add monorepo command parity

Update root `Makefile`:

- add `test-form-service` target (`cd services/form && uv run pytest -v --cov=app`)
- include target in `.PHONY`
- include command in `make help`

### 5) Add dedicated CI workflow parity

Create `.github/workflows/test-form-service.yaml` analogous to:

- `.github/workflows/test-mail-service.yaml`
- `.github/workflows/test-user-service.yaml`

It should:

- trigger on changes under `services/form/**`
- set up Python 3.14 + uv
- run `uv sync --all-extras` in `services/form`
- run ruff checks for form service
- run `make test-form-service`

### 6) Update service docs

Update `services/form/README.md` with a **Testing** section that documents:

- how to run form tests locally
- expected command (`make test-form-service`)

## Acceptance Criteria

- [ ] `services/form/pyproject.toml` includes dev/test extras and pytest config matching backend conventions
- [ ] `services/form/tests/` exists with `conftest.py` and core test modules
- [ ] `make test-form-service` exists and runs form tests with coverage
- [ ] `.github/workflows/test-form-service.yaml` exists and runs on form changes
- [ ] `services/form/README.md` documents form testing command(s)
- [ ] New test suite covers core form + submission happy/error flows
- [ ] CI passes for the new form-service test workflow

## Non-Goals

- Large refactors of form business logic unrelated to testing
- Rewriting existing mail/user tests
- End-to-end cross-service integration tests in this first step

## Notes

Suggested rollout:

1. land test infrastructure + minimal critical tests
2. add deeper branch coverage incrementally (DB error paths, edge-case validation, time-boundary cases)

This keeps delivery fast while immediately bringing `form` to the same test workflow baseline as other backend services.

### PR DESCRIPTION
**Pull request by poBogan**
## What does this PR do?

Add comprehensive pytest suite for the form service (API, models, services) including fixtures and test helpers; mock MongoDB interactions for fast unit tests. Add GitHub Actions workflow (.github/workflows/test-form-service.yaml) to run tests on pushes/PRs for services/form using Python 3.14 and the uv tool. Add Makefile target (test-form-service) and update services/form/pyproject.toml with email-validator, dev dependencies (pytest, pytest-asyncio, pytest-cov, ruff, httpx, mongomock-motor) and pytest config. Update services/form README with testing instructions.

## Related Issue

Closes #88 

## Checklist

- [x] Code follows project conventions
- [ ] I tested my changes locally
- [x] Linting passes
- [x] I added @seberatolmez or @dogukanurker as reviewers

### CODE REVIEW

**Review by DogukanUrker — CHANGES_REQUESTED** [review]
Revert your changes in makefile about uv installation. Uv is a repo level dependency so every developer in the repo MUST install all the dependencies before writing any single line of code. If you dont have uv and this is why you can not run the tests, this is not our/repo level problem, this is your problem. So please be sure you have all the dependencies installed in your system before writing code. And test your prs e2e to be sure everything is working. By today we pushed some of our services to prod, and what its mean is our changes will directly effect the prod when we merge our stuff. So we must test everything manually and with the test flow

**Comment by DogukanUrker** [comment]
Also please mark the "i test my changes locally" checkbox in the pr description. These checkboxes are really helpful for reviewers to understand the status of the pr

File: .gitignore [hunk 1]
```diff
@@ -29,8 +29,6 @@ dist/
 .env
 .env.local
 .env.*.local
-.env.prod
-.env.*.prod
```
Comment (DogukanUrker) [inline .gitignore]: Why this change? I think you made something wrong while rebasing your branch. Cause i added this things today to main and we need them

**Review by DogukanUrker — CHANGES_REQUESTED** [review]
Fix broken rebase please check the pr diff you are reverting sooo many changes which we have added today
